src/idiomorph.js

```diff
@@ -1063,11 +1074,48 @@ var Idiomorph = (function () {
         function removeNode(tempNode, ctx) {
             removeIdsFromConsideration(ctx, tempNode)
             if (ctx.callbacks.beforeNodeRemoved(tempNode) === false) return;
-
-            tempNode.parentNode?.removeChild(tempNode);
+            if (ctx.pantry) {
+                moveToPantry(tempNode, ctx);
+            } else {
+                tempNode.parentNode?.removeChild(tempNode);
+            }
             ctx.callbacks.afterNodeRemoved(tempNode);
         }
 
+        function moveToPantry(node, ctx) {
+            // If the node is a leaf (no children), process it, and then we're done
+            if (!node.children || node.children.length === 0) {
+                if (node.id) {
+                    ctx.pantry.appendChild(node);
+                }
+
+            // otherwise we need to process the children first
+            } else {
+                Array.from(node.children).forEach(child => {
+                    moveToPantry(child, ctx);
+                });
+
+                // After processing children, process the current node
+                if (node.id) {
+                    node.innerHTML = '';
+                    ctx.pantry.appendChild(node);
+                }
+            }
+        }
+
+        function restoreFromPantry(root, ctx) {
+            Array.from(ctx.pantry.children).forEach(element => {
+                const matchElement = root.querySelector(`#${element.id}`);
+                if (matchElement) {
+                    matchElement.before(element);
+                    element.replaceChildren(matchElement.childNodes);
```

Comment:
I found in my testing that this was adding invalid child node data but adding a ... spread operator fixed the typing for me